src/System.Security.Claims/tests/ClaimTests.cs

```diff
@@ -14,5 +15,29 @@ public void Ctor_ArgumentValidation()
         {
             Assert.Throws<ArgumentNullException>(() => new Claim(null));
         }
+
+        [Fact]
+        public void BinaryWriteReadTest_Success()
+        {
+            var claim = new Claim(ClaimTypes.Actor, "Test", ClaimValueTypes.String, "Test", "Test")
+            claim.Properties["Prop1"] = "Value1";
+            claim.Properties["Prop2"] = "Value2";
+
+            using (var memoryStream = new MemoryStream())
+            {
+                using (var binaryWriter = new BinaryWriter(memoryStream, Encoding.Default, true))
+                {
+                    claim.WriteTo(binaryWriter);
+                    binaryWriter.Flush();
+                }
+
+                memoryStream.Position = 0;
+                using (var binaryReader = new BinaryReader(memoryStream))
+                {
+                    var clonedClaim = new Claim(binaryReader);
+                    Assert.NotNull(clonedClaim);
```

Comment:
This assert isn't useful.  There'd have to be a serious bug in the runtime for `new` to return null.